### ISSUE TITLE
New Stuff

### ISSUE THREAD

**Issue by mzuenni** [mzuenni/ChaosKITs#1]
These Topics seem useless:

- 4.14 Primzahlzählfunktion π
- 4.11 Faktorisierung squfof

Do we want to replace this with new stuff? (maybe more geometry?) @Noobie99

**Comment by imirdy** [comment]
I don't think we will ever use 4.14. I don't really know what 4.11 squof does, is there any reason to ever use it over rho, since it looks like it can fail?

We can probably replace those, but I don't really know what to put there instead. Is there anything you have in mind?

**Comment by mzuenni** [comment]
It is a factorisation algorithm with better constant factors then rho... but i dont think that it will ever be useful...
I dont have much ideas for new stuff right now... maybe:

- [ ] half plane intersection
- [x] a better treap? @ammen99
- [x] Polygon line Intersection in log(n)
- [x] SOS dp 
- [ ] ...?

**Comment by mzuenni** [comment]
2.17.1 Push Relabel
seems to be unneccessary as well imo (and i am not even sure if the implmementation is correct?)

**Comment by imirdy** [comment]
It seems correct (and pretty fast) but still probably unnecessary.

**Comment by mzuenni** [comment]
This might also be useful: https://codeforces.com/blog/entry/112458
